File: Samples/VB/15-ContextPopup/Form1.vb
Imports RibbonLib
Imports RibbonLib.Controls
Imports RibbonLib.Interop

Namespace _15_ContextPopup
	Public Enum RibbonMarkupCommands As UInteger
		cmdButtonNew = 1001
		cmdButtonOpen = 1002
		cmdButtonSave = 1003
		cmdButtonExit = 1004
		cmdContextMap = 1005
		cmdFontControl = 1006
		cmdDropDownColorPicker = 1007
	End Enum

	Partial Public Class Form1
		Inherits Form
		Private _ribbonContextMenuStrip As RibbonContextMenuStrip

		Public Sub New()
			InitializeComponent()

			_ribbonContextMenuStrip = New RibbonContextMenuStrip(_ribbon, CUInt(RibbonMarkupCommands.cmdContextMap))

			' recommended way
			AddHandler panel2.MouseClick, AddressOf panel2_MouseClick

			' convenient way
			panel1.ContextMenuStrip = _ribbonContextMenuStrip
		End Sub

		Private Sub panel2_MouseClick(ByVal sender As Object, ByVal e As MouseEventArgs)
			If e.Button = MouseButtons.Right Then
				Dim p As Point = panel2.PointToScreen(e.Location)
				_ribbon.ShowContextPopup(CUInt(RibbonMarkupCommands.cmdContextMap), p.X, p.Y)
			End If
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			_ribbon.Viewable = False
		End Sub
	End Class
End Namespace

File: Samples/CS/15-ContextPopup/Form1.cs
using System;
using System.Windows.Forms;

using RibbonLib;
using RibbonLib.Controls;
using RibbonLib.Interop;

namespace _15_ContextPopup
{
    public enum RibbonMarkupCommands : uint
    {
        cmdButtonNew = 1001,
        cmdButtonOpen = 1002,
        cmdButtonSave = 1003,
        cmdButtonExit = 1004,
        cmdContextMap = 1005,
        cmdFontControl = 1006,
        cmdDropDownColorPicker = 1007,
    }
    
    public partial class Form1 : Form
    {
        private RibbonContextMenuStrip _ribbonContextMenuStrip;

        public Form1()
        {
            InitializeComponent();

            _ribbonContextMenuStrip = new RibbonContextMenuStrip(_ribbon, (uint)RibbonMarkupCommands.cmdContextMap);

            // recommended way
            panel2.MouseClick += new MouseEventHandler(panel2_MouseClick);

            // convenient way
            panel1.ContextMenuStrip = _ribbonContextMenuStrip;
        }

        void panel2_MouseClick(object sender, MouseEventArgs e)
        {
            if (e.Button == MouseButtons.Right)
            {
                System.Drawing.Point p = panel2.PointToScreen(e.Location);
                _ribbon.ShowContextPopup((uint)RibbonMarkupCommands.cmdContextMap, p.X, p.Y);
            }
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            _ribbon.Viewable = false;
        }
    }
}
